sunspec.go

```diff
@@ -28,10 +28,125 @@ type ScaleFactor int16
 
 type Count uint16
 
+// An array is a collection of devices.
+//
+// This type is used when the underlying representation
+// is an XML document.
+type Array interface {
+	Device(id string) Device // Answer the specified device instance or nil
+	Do(func(d Device))       // Iterate over all the devices.
+}
```

Comment:
The purpose of this type is to model the collection of devices that can appear in an XML document. 

I chose the name "Array" based more on usage in "Allen Telescope Array" rather than the traditional computer science meaning of array although, of course, both are related uses.